docs/proposals/tenant_retention.md

```diff
@@ -0,0 +1,64 @@
+---
+title: "Retention of Tenant Data from Blocks Storage"
+linkTitle: "Retention of Tenant Data from Blocks Storage"
+weight: 1
+slug: tenant-retention
+---
+
+- Author: [Allenzhli](https://github.com/Allenzhli)
+- Date: December 2020
+- Status: Proposed
+
+## Retention of tenant data
+
+## Problem
+
+In Cortex, we are missing retention of blocks storage. The metric data is growing over time per-tenant, at the same time, the value of data decreases. We want to support retention time of metric data like prometheus does.
+
+## Background
+
+When using blocks storage, Cortex stores tenant’s data in object store for long-term storage of blocks, tenant id as part of the object store path. We discover all tenants via scan the root dir of bucket.
+
+## Proposal
+
+### API Endpoints
+
+#### POST /compactor/retention_tenant
+
+We propose to introduce an `/compactor/retention_tenant` API endpoint to set data retention for the tenant.
```

Comment:
I wouldn't introduce any new API. We should use the already existing mechanism used to set per-tenant overrides (via the runtime config "overrides").

Reply:
I will delete the api design.